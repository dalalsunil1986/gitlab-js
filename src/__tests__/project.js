jest.mock('../utils/api', () => jest.fn());
jest.mock('extra-log', () => ({ msg: jest.fn() }));

const project = require('../project');
const api = require('../utils/api');

describe('request', () => {
  beforeEach(() => {
    api.mockReset();
    api.mockImplementation((...params) => [...params]);
  });


  it('all', async () => {
    const response = await project.all();

    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(1);
    expect(api.mock.calls).toMatchSnapshot();
  });


  it('all with opts', async () => {
    const response = await project.all({ simple: false, owned: true });

    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(1);
    expect(api.mock.calls).toMatchSnapshot();
  });


  it('owned', async () => {
    const response = await project.owned('userId');

    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(1);
    expect(api.mock.calls).toMatchSnapshot();
  });


  it('owned, no params', async () => {
    const response = await project.owned();

    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(0);
  });
});
