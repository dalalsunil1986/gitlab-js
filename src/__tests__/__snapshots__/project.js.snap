// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request all 1`] = `
Array [
  "/projects",
  Object {
    "simple": true,
  },
]
`;

exports[`request all 2`] = `
Array [
  Array [
    "/projects",
    Object {
      "simple": true,
    },
  ],
]
`;

exports[`request all with opts 1`] = `
Array [
  "/projects",
  Object {
    "owned": true,
    "simple": false,
  },
]
`;

exports[`request all with opts 2`] = `
Array [
  Array [
    "/projects",
    Object {
      "owned": true,
      "simple": false,
    },
  ],
]
`;

exports[`request owned 1`] = `
Array [
  "/users/userId/projects",
  Object {
    "simple": true,
  },
]
`;

exports[`request owned 2`] = `
Array [
  Array [
    "/users/userId/projects",
    Object {
      "simple": true,
    },
  ],
]
`;

exports[`request owned, no params 1`] = `undefined`;
